File: frontend/src/api/dictionaries.ts
import { api } from "./client";

export interface Program {
  id: string;
  name: string;
  code: string;
  is_active: boolean;
}

export interface Specialty {
  id: string;
  name: string;
  code: string;
  is_active: boolean;
  program_ids: string[];
}

export interface Department {
  id: string;
  name: string;
  code: string;
  is_active: boolean;
}

export interface Cohort {
  id: string;
  name: string;
  start_date: string;
  end_date: string;
  is_active: boolean;
}

export async function getPrograms() {
  return api<Program[]>("/dictionaries/programs?active=true");
}

export async function getSpecialties() {
  return api<Specialty[]>("/dictionaries/specialties?active=true");
}

export async function getDepartments() {
  return api<Department[]>("/dictionaries/departments?active=true");
}

export async function getCohorts() {
  return api<Cohort[]>("/dictionaries/cohorts?active=true");
}

File: backend/internal/models/dictionaries.go
package models

// Program moved to curriculum.go
// Cohort moved to curriculum.go


type Specialty struct {
	ID         string   `db:"id" json:"id"`
	Name       string   `db:"name" json:"name"`
	Code       string   `db:"code" json:"code"`
	ProgramIDs []string `json:"program_ids"` // Multiple programs
	IsActive   bool     `db:"is_active" json:"is_active"`
	CreatedAt  string   `db:"created_at" json:"created_at"`
	UpdatedAt  string   `db:"updated_at" json:"updated_at"`
}

type Department struct {
	ID        string `db:"id" json:"id"`
	Name      string `db:"name" json:"name"`
	Code      string `db:"code" json:"code"`
	IsActive  bool   `db:"is_active" json:"is_active"`
	CreatedAt string `db:"created_at" json:"created_at"`
	UpdatedAt string `db:"updated_at" json:"updated_at"`
}
